netsim/modules/bgp.py

```diff
@@ -269,31 +267,52 @@ def build_ebgp_sessions(node: Box, sessions: Box, topology: Box) -> None:
 
       # Figure out whether both neighbors have IPv6 LLA and/or unnumbered IPv4 interfaces
       #
-      ipv6_lla = l.get('ipv6',None) is True and ngb_ifdata.get('ipv6',None) is True
-      ipv6_num = isinstance(l.get('ipv6',None),str) and isinstance(ngb_ifdata.get('ipv6',None),str)
-      rfc8950  = l.get('unnumbered',None) is True and ngb_ifdata.get('unnumbered',None) is True
-      ipv4_unnum = l.get('ipv4',None) is True and ngb_ifdata.get('ipv4',None) is True
-#      print(f'EBGP node {node.name} neighbor {ngb_name} lla {ipv6_lla} v6num {ipv6_num} v4unnum {ipv4_unnum} rfc8950 {unnumbered}')
-      if ipv4_unnum and not ipv6_num:                                 # Unnumbered IPv4 w/o IPv6 ==> IPv6 LLA + RFC 8950 IPv4 AF
-        rfc8950 = True
-
-#      print(f'... unnumbered {unnumbered}')
-      if ipv6_lla or rfc8950:
-        extra_data.local_if = l.ifname                                # Set local_if to indicate IPv6 LLA EBGP session
-        if not features.bgp.ipv6_lla:                                 # IPv6_LLA feature flag has to be set even for IPv4 unnumbered EBGP
+      ipv4_addr = l.get('ipv4',None)
+      ipv6_addr = l.get('ipv6',None)
+      ipv6_lla = ipv6_addr is True and ngb_ifdata.get('ipv6',None) is True
+      ipv6_num = isinstance(ipv6_addr,str) and isinstance(ngb_ifdata.get('ipv6',None),str)
+      ipv4_unnum = ipv4_addr is True and ngb_ifdata.get('ipv4',None) is True
+      rfc8950 = ipv4_unnum and ipv6_num                               # RFC 8950 IPv4 next hops over non-LLA EBGP
+
+      # Sanity checks: both ends have to be numbered or unnumbered
+      #
+      if ipv4_addr is True and not ipv4_unnum:
+        log.error(
+          f'Cannot create an EBGP session between IPv4 unnumbered interface on {node.name}' + \
+          f' and regular IPv4 interface on {ngb_name}',
+          category=log.IncorrectValue,
+          module='bgp')
+        continue
+
+      if ipv6_addr is True and not ipv6_lla:
+        log.error(
+          f'Cannot create an EBGP session between IPv6 LLA-only interface on {node.name}' + \
+          f' and numbered IPv6 interface on {ngb_name}',
+          category=log.IncorrectValue,
+          module='bgp')
+        continue
+
+      if ipv4_unnum and not ipv6_num:                 # Unnumbered IPv4 without numbered IPv6
+        ipv6_lla = True                               # ... requires IPv6 LLA session
+        if not 'ipv6' in l:                           # ... and IPv6 enabled on the interface
+          l.ipv6 = True
```

Comment:
Do we need to update the neighbor too?

Reply:
No. The same code is executed on the neighbor node.